addons/analytic/models/analytic_mixin.py

```diff
@@ -87,6 +87,56 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
         domain = self._apply_analytic_distribution_domain(domain)
         return super().read_group(domain, fields, groupby, offset, limit, orderby, lazy)
 
+    def mapped(self, key):
+        """
+            Change analytic_distribution mapped value used in filtered_domain
+            to a list of keys instead of having the whole “json” object.
+            It's easier to compare ids with ids than ids with objects
+
+            example:
+            before: self.mapped("analytic_distribution") => [{"16,17": 100.0}]
+            after: self.mapped("analytic_distribution") => [16, 17]
+        """
+        if key == 'analytic_distribution' and self.env.context.get('distribution_ids'):
+            return self._get_analytic_account_ids()
```

Comment:
`mapped` needs to work for multiple records (or none); even if it is always exactly one record inside `filtered_domain`
```suggestion
            return [record._get_analytic_account_ids() for record in self]
```

Just FYI, I think another solution could be
```suggestion
            key = self.pool[self._name]._get_analytic_account_ids
```